### PR TITLE
Added support for escaped CRLF newlines in Makefile Parser

### PR DESCRIPTION
**Pull request by gmittert**

### CODE REVIEW

**Comment by gmittert** [comment]
@swift-ci please smoke test

File: lib/Core/MakefileDepsParser.cpp [hunk 1]
```diff
@@ -70,6 +70,12 @@ static void skipNonNewlineWhitespace(const char*& cur, const char* end) {
       continue;
     }
 
+    // Also skip \r\n newlines
+    if (c == '\\' && cur + 2 != end && cur[1] == '\r' && cur[2] == '\n') {
```
Comment (dmbryson) [inline lib/Core/MakefileDepsParser.cpp]: I think that needs to be something like `cur + 2 < end`.  If we have only 1 byte left, the +2 would miss that we're going to read past the end.

Comment (dmbryson) [reply lib/Core/MakefileDepsParser.cpp]: One way to show that would be to add a test for `input = "out: \\\r";`

Comment (gmittert) [reply lib/Core/MakefileDepsParser.cpp]: Good point, fixed.

**Comment by gmittert** [comment]
@swift-ci please smoke test

**Comment by gmittert** [comment]
@swift-ci please smoke test

**Pull request merged**